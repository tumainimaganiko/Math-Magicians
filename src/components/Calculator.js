import { useState } from 'react';
import Span from './Span';

export default function Calculator() {
  const [obj, setObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  return (
    <>
      <h2>Let&apos;s do some Math</h2>
      <div className="container">
        <div className="display">{obj.next || obj.total || '0'}</div>
        <Span obj={obj} setObj={setObj} text="AC" color="white" />
        <Span obj={obj} setObj={setObj} text="+/-" color="white" />
        <Span obj={obj} setObj={setObj} text="%" color="white" />
        <Span obj={obj} setObj={setObj} text="รท" color="yellow" />
        <Span obj={obj} setObj={setObj} text="7" color="white" />
        <Span obj={obj} setObj={setObj} text="8" color="white" />
        <Span obj={obj} setObj={setObj} text="9" color="white" />
        <Span obj={obj} setObj={setObj} text="x" color="yellow" />
        <Span obj={obj} setObj={setObj} text="4" color="white" />
        <Span obj={obj} setObj={setObj} text="5" color="white" />
        <Span obj={obj} setObj={setObj} text="6" color="white" />
        <Span obj={obj} setObj={setObj} text="-" color="yellow" />
        <Span obj={obj} setObj={setObj} text="1" color="white" />
        <Span obj={obj} setObj={setObj} text="2" color="white" />
        <Span obj={obj} setObj={setObj} text="3" color="white" />
        <Span obj={obj} setObj={setObj} text="+" color="yellow" />
        <Span obj={obj} setObj={setObj} text="0" color="zero white" />
        <Span obj={obj} setObj={setObj} text="." color="white" />
        <Span obj={obj} setObj={setObj} text="=" color="yellow" />
      </div>
    </>
  );
}
