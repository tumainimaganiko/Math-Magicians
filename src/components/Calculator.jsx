import { useState } from "react";
import styles from "styles/Calculator.module.css";
import Span from "./Span";

export default function Calculator() {
  const [obj, setObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const heading = {
    padding: "20px",
  };

  return (
    <div className={styles.container}>
      <h2 style={heading}>Let&apos;s do some Math!</h2>
      <div className="container">
        <div className="display">{obj.next || obj.total || "0"}</div>
        <Span obj={obj} setObj={setObj} text="AC" color="white" />
        <Span obj={obj} setObj={setObj} text="+/-" color="white" />
        <Span obj={obj} setObj={setObj} text="%" color="white" />
        <Span obj={obj} setObj={setObj} text="รท" color="yellow" />
        <Span obj={obj} setObj={setObj} text="7" color="white" />
        <Span obj={obj} setObj={setObj} text="8" color="white" />
        <Span obj={obj} setObj={setObj} text="9" color="white" />
        <Span obj={obj} setObj={setObj} text="x" color="yellow" />
        <Span obj={obj} setObj={setObj} text="4" color="white" />
        <Span obj={obj} setObj={setObj} text="5" color="white" />
        <Span obj={obj} setObj={setObj} text="6" color="white" />
        <Span obj={obj} setObj={setObj} text="-" color="yellow" />
        <Span obj={obj} setObj={setObj} text="1" color="white" />
        <Span obj={obj} setObj={setObj} text="2" color="white" />
        <Span obj={obj} setObj={setObj} text="3" color="white" />
        <Span obj={obj} setObj={setObj} text="+" color="yellow" />
        <Span obj={obj} setObj={setObj} text="0" color="zero white" />
        <Span obj={obj} setObj={setObj} text="." color="white" />
        <Span obj={obj} setObj={setObj} text="=" color="yellow" />
      </div>
    </div>
  );
}
