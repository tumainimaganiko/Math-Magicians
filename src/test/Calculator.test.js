import renderer from 'react-test-renderer';
import Calculator from 'components/Calculator';

it('Calculate', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
