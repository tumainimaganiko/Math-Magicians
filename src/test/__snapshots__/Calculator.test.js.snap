// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculate 1`] = `
<div
  className="container"
>
  <h2
    style={
      Object {
        "padding": "20px",
      }
    }
  >
    Let's do some Math!
  </h2>
  <div
    className="container"
  >
    <div
      className="display"
    >
      0
    </div>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="yellow"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="yellow"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="yellow"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="yellow"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero white"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="white"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="yellow"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
